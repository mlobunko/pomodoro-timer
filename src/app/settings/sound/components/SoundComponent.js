import React from "react";
import PropTypes from "prop-types";

import Button from "shared/button/components";

export const SoundComponent = ({ isSound, change }) => (
  <React.Fragment>
    <div>Rest</div>
    <div>
      <Button
        round
        icon={isSound ? "volumeUp" : "volumeOff"}
        onClick={change}
      />
    </div>
  </React.Fragment>
);

SoundComponent.displayName = "Sound";

export default SoundComponent;

SoundComponent.propTypes = {
  isSound: PropTypes.bool.isRequired,
  change: PropTypes.func.isRequired
};
