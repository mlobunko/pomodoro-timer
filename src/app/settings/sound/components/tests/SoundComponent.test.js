import React from "react";
import { shallow } from "enzyme";

import { SoundComponent } from "../SoundComponent";

const props = {
  change: jest.fn(),
  round: true,
  isSound: true
};

describe("props", () => {
  it("button onClick is 'change'", () => {
    const wrapper = shallow(<SoundComponent {...props} />);
    expect(wrapper.find("[round=true]").prop("onClick")).toEqual(props.change);
  });

  it("button icon is 'volumeUp'", () => {
    const wrapper = shallow(<SoundComponent {...props} />);
    expect(wrapper.find("[round=true]").prop("icon")).toEqual("volumeUp");
  });

  it("button icon is 'volumeOff'", () => {
    const wrapper = shallow(<SoundComponent {...props} isSound={false} />);
    expect(wrapper.find("[round=true]").prop("icon")).toEqual("volumeOff");
  });
});

describe("render", () => {
  it("renders correctly, isSound true", () => {
    const wrapper = shallow(<SoundComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, isSound false", () => {
    const wrapper = shallow(<SoundComponent {...props} isSound={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
