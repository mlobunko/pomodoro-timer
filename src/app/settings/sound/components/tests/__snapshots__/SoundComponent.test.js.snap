// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly, isSound false 1`] = `
<Fragment>
  <div>
    Rest
  </div>
  <div>
    <Button
      icon="volumeOff"
      onClick={[Function]}
      round={true}
    />
  </div>
</Fragment>
`;

exports[`render renders correctly, isSound true 1`] = `
<Fragment>
  <div>
    Rest
  </div>
  <div>
    <Button
      icon="volumeUp"
      onClick={[Function]}
      round={true}
    />
  </div>
</Fragment>
`;
